src/Atmos/Model/source.jl

```diff
@@ -182,6 +183,43 @@ function atmos_source!(
     end
 end
 
+"""
+    CreateClouds{FT} <: Source
+
+A source/sink to `q_liq` and `q_ice` implemented as a relaxation towards
+equilibrium in the Microphysics module.
+The default relaxation timescales are defined in CLIMAParameters.jl.
+"""
+struct CreateClouds <: Source end
+function atmos_source!(
+    ::CreateClouds,
+    atmos::AtmosModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+)
+    # get current temperature and phase partition
+    FT = eltype(state)
+    ts = recover_thermo_state(atmos, state, aux)
+    q = PhasePartition(ts)
+    T = air_temperature(ts)
+
+    # phase partition corresponding to the current T and q.tot
+    # (this is not the same as phase partition from saturation adjustment)
+    ts_eq = TemperatureSHumEquil(atmos.param_set, T, state.ρ, q.tot)
+    q_eq = PhasePartition(ts_eq)
```

Comment:
Now I understand 😄. Looks good!

Reply:
I had to re-understand it myself!

Reply:
Not now, but perhaps we can write a `new_thermo_state_eq` convenience wrapper in the case that equilibrium states/phase partitioning is frequently needed in the NonEquil components.

Reply:
Agreed. Maybe once I add the precipitation interface and all the microphysics source terms are integrated, we can reevaluate what wrappers we need. It will be more clear what functions are used and to what purpose.